src/components/Modals/Snaps/helpers.ts

```diff
@@ -0,0 +1,57 @@
+import { type ChainId, fromChainId } from '@shapeshiftoss/caip'
+import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import { deriveAccountIdsAndMetadataForChainNamespace } from 'lib/account/account'
+import { portfolio, portfolioApi } from 'state/slices/portfolioSlice/portfolioSlice'
+import type { AppDispatch } from 'state/store'
+
+export const fetchAccountForChainId = async (
```

Comment:
And similar to its consumption - assuming actual automagical discovery (last happy commit: cfd1e15c59), we shouldn't need this